tests/test_ote_training.py

```diff
@@ -0,0 +1,690 @@
+    name = score.name
+    value = score.value
+    return name, value
+
+def select_configurable_parameters(json_configurable_parameters):
+    selected = {}
+    getv = lambda c, n: c[n]['value']
+    for section, container in json_configurable_parameters.items():
+        for param in container.keys():
+            try:
+                selected[f'{section}/{param}'] = getv(container, param)
+            except TypeError:
+                pass
+            except KeyError:
+                pass
+    return selected
+
+def convert_hyperparams_to_dict(hyperparams):
+    def _convert(p):
+        if p is None:
+            return None
+        d = {}
+        groups = getattr(p, 'groups', [])
+        parameters = getattr(p, 'parameters', [])
+        assert (not groups) or isinstance(groups, list), f'Wrong field "groups" of p={p}'
+        assert (not parameters) or isinstance(parameters, list), f'Wrong field "parameters" of p={p}'
+        for group_name in groups:
+            g = getattr(p, group_name, None)
+            d[group_name] = _convert(g)
+        for par_name in parameters:
+            d[par_name] = getattr(p, par_name, None)
+        return d
+    return _convert(hyperparams)
+
+class OTETrainingImpl:
+    def __init__(self, dataset_params: DatasetParameters, template_file_path: str,
+                 num_training_iters: int):
+        self.dataset_params = dataset_params
+        self.template_file_path = template_file_path
+        self.num_training_iters = num_training_iters
+
+        self.template = None
+        self.environment = None
+        self.task = None
+        self.output_model = None
+        self.evaluation_performance = None
+        self.environment_for_export = None
+        self.exported_model = None
+        self.openvino_task = None
+        self.evaluation_performance_exported = None
+
+        self.was_training_run = False
+        self.stored_exception_training = None
+        self.was_export_run = False
+        self.stored_exception_export = None
+
+        self.copy_hyperparams = None
+
+    @staticmethod
+    def _create_environment_and_task(params, labels_schema, model_template):
+        environment = TaskEnvironment(model=None, hyper_parameters=params, label_schema=labels_schema,
+                                      model_template=model_template)
+        logger.info('Create base Task')
+        task_impl_path = model_template.entrypoints.base
+        task_cls = get_task_class(task_impl_path)
+        task = task_cls(task_environment=environment)
+        return environment, task
+
+    def _run_ote_training(self):
+        print(f'self.template_file_path = {self.template_file_path}')
+        print(f'Using for train annotation file {self.dataset_params.annotations_train}')
+        print(f'Using for val annotation file {self.dataset_params.annotations_val}')
+
+
+        self.dataset = MMDatasetAdapter(
+            train_ann_file=self.dataset_params.annotations_train,
```

Comment:
You can remove your named tuple and use dict, and unpack self.dataset_params passing it here:
MMDatasetAdapter(**self.dataset_params, dataset_storage=NullDatasetStorage())

Reply:
It is possible, but I'm not sure in `dataset_params` field -- it is received outside the class. I prefer to list the fields explicitely.

Reply:
How it can be tested locally? Especially for users who don't access to our datasets?